probe-rs-tools/src/bin/probe-rs/cmd/dap_server/server/debugger.rs

```diff
@@ -723,12 +745,31 @@ impl Debugger {
 
         download_options.progress = flash_progress;
 
-        let loader = build_loader(
+        let loader = match build_loader(
             &mut session_data.session,
             path_to_elf,
             self.config.flashing_config.format_options.clone(),
             None,
-        )?;
+        ) {
+            Ok(loader) => loader,
+            Err(error) => {
+                drop(download_options);
```

Comment:
This could use a comment. My guess is that `download_options` holds onto `debug_adapter`, but that isn't obvious and random drops are always a bit on the "can I remove it?" side.

Reply:
Done :)